cmd/helm/upgrade.go

```diff
@@ -91,9 +91,11 @@ func newUpgradeCmd(cfg *action.Configuration, out io.Writer) *cobra.Command {
 
 			if client.Install {
 				// If a release does not exist, install it. If another error occurs during
-				// the check, ignore the error and continue with the upgrade.
```

Comment:
@bacongobbler I just saw this comment again.  It explicitly says to ignore other errors.  It was written by @technosophos back in 2016 in 36e6094.  Maybe it was because of some tiller need?  Do you feel it is safe to stop ignoring other errors?

Reply:
This code was originally executed in Tiller. Back then, the history client returned only a small set of errors, and the only one we cared about was ErrReleaseNotFound. If that happened, we knew we _could_ create a new install. Any other error case, we didn't care about -- we would just press on with the upgrade and see if we could make more progress. At the time, any other errors coming out of histClient.Run did not seem to warrant aborting the entire upgrade effort.

Your change makes a lot of sense now that Tiller is gone and connecting to the API server is more volatile. It is possible that we may catch a spurious error that wouldn't actually stop a successful upgrade, but I'd have to read the current Run code to know if that's the case.

Reply:
Great. Thanks for the explanation. So we'll go with @bacongobbler 's suggestion of aborting on all errors.

Reply:
I'll push a new version removing the comment.